ts/src/pro/poloniexfutures.ts

```diff
@@ -0,0 +1,741 @@
+        //
+        const data = this.safeValue (message, 'data', []);
+        // const type = this.safeString (message, 'subject');
+        const marketId = this.safeString (data, 'symbol');
+        const market = this.safeMarket (marketId);
+        // const orderId = this.safeString (data, 'orderId');
+        const timestamp = this.safeInteger (data, 'ts') / 1000;
+        const messageHash = this.safeString (data, 'topic');
+        const side = this.safeString (data, 'side');
+        const size = this.safeString (data, 'size');
+        const price = this.safeString (data, 'price');
+        const symbol = this.safeString (market, 'symbol');
+        const subscription = this.safeValue (client.subscriptions, messageHash, {});
+        const limit = this.safeInteger (subscription, 'limit');
+        // const update = type === 'done';
+        const orderBook = this.safeValue (this.orderbooks, symbol);
+        if (orderBook === undefined) {
+            this.orderbooks[symbol] = this.orderBook ({}, limit);
+        }
+        if (side === 'buy') {  // Only happens if subject is open
+            orderBook['bids'].store (price, size);
+        } else if (side === 'sell') {
+            orderBook['asks'].store (price, size);
+        }
+        // TODO: How to remove is subject is done? Is only possible with orderId
+        orderBook['timestamp'] = timestamp;
+        orderBook['datetime'] = this.iso8601 (timestamp);
+        client.resolve (orderBook, messageHash);
+    }
+
+    handleBalance (client, message) {
+        //
+        //    {
+        //        data: {
+        //          currency: 'USDT',
+        //          availableBalance: '4.0000000000',
+        //          timestamp: '1680557568670'
+        //        },
+        //        subject: 'availableBalance.change',
+        //        topic: '/contractAccount/wallet',
+        //        channelType: 'private',
+        //        id: '642b4600cae86800074b5ab7',
+        //        type: 'message',
+        //        userId: '1139790'
+        //    }
+        //
+        //    {
+        //        data: {
+        //          currency: 'USDT',
+        //          orderMargin: '0.0000000000',
+        //          timestamp: '1680558743307'
+        //        },
+        //        subject: 'orderMargin.change',
+        //        topic: '/contractAccount/wallet',
+        //        channelType: 'private',
+        //        id: '642b4a97b58e360007c3a237',
+        //        type: 'message',
+        //        userId: '1139790'
+        //    }
+        //
+        const data = this.safeValue (message, 'data', []);
+        const messageHash = '/contractAccount/wallet';
+        const currencyId = this.safeString (data, 'currency');
+        const currency = this.currency (currencyId);
+        const code = currency['code'];
+        const balance = this.safeValue (this.balance, code, {});
+        this.balance[code] = this.deepExtend (balance, this.parseWsBalance (data));
```

Comment:
I'm having some trouble with whats returned from the exchange when an order is placed. 2 response objects will come separately, one saying how much margin is available, and one saying how much is used. Here I try merging them together so that the balance stored contains both, but with no success, I hoped to have `free` and `used` stored inside `this.balance[code]` but it looks like only the most recent response is being stored

```
poloniexfutures.watchBalance ()
{
  info: {
    currency: 'USDT',
    availableBalance: '4.5007500000',
    timestamp: '1680730764910'
  },
  timestamp: 1680730764910,
  datetime: '2023-04-05T21:39:24.910Z',
  USDT: { free: 4.50075, used: undefined, total: undefined },
  free: { USDT: 4.50075 },
  used: { USDT: undefined },
  total: { USDT: undefined }
}
{
  info: {
    currency: 'USDT',
    availableBalance: '4.5007500000',
    timestamp: '1680730764910',
    orderMargin: '2.5037500000'
  },
  timestamp: 1680730764910,
  datetime: '2023-04-05T21:39:24.910Z',
  USDT: { free: undefined, used: 2.50375, total: undefined },
  free: { USDT: undefined },
  used: { USDT: 2.50375 },
  total: { USDT: undefined }
}
```

Reply:
Here in `parseWsBalance` you are creating a new account on every message and thats why it's only getting the last value. You could create a new account only if it doesn't exist and then update the corresponding value.

However I would be careful using deep Extend here as we may join old data with new and give an incorrect answer.
Maybe a solution would be to check the timestamp, and if they are the same then merge them?